### ISSUE TITLE
Incorrect handling of `..` in root directory on php 5.5

### ISSUE THREAD

**Issue by acoulton** [mikey179/vfsStream#115]
We're migrating a project from PHP 5.3 to 5.5 and the unit tests for a garbage collector class have just started failing.

It seems stat information for `/..` is loaded incorrectly, so it is not marked as a directory and attempting to access the mtime causes an exception.

This script:
```php
require_once(__DIR__.'/vendor/autoload.php');
error_reporting(E_ALL | E_STRICT);
ini_set('display_errors', 0);

$vfs = vfsStream::setup('root');
vfsStream::create(['tmp' => []]);

$cases = [
	'Real root file system'   => '/',
	'Real nested file system' => '/tmp',
    'VFS root file system' => vfsStream::url('root'),
    'VFS nested file system' => vfsStream::url('root/tmp'),
];

foreach ($cases as $caption => $root_path)
{
	print "\n\n";
	print "$caption\n";
	print str_repeat('-', strlen($caption))."\n";
	$iterator = new DirectoryIterator($root_path);
	foreach ($iterator as $file)
	{
		if (!in_array($file->getFilename(), ['.', '..']))
		{
			continue;
		}

		print sprintf(
			"%-10s - isDir:%b isDot:%b isFile:%b mtime:",
			$file->getPathname(),
			$file->isDir(),
			$file->isDot(),
			$file->isFile()
		);

		try
		{
			print $file->getMTime()."\n";
		} catch (\Exception $e)
		{
			print "?\nERROR: ".$e->getMessage()."\n";
		}
	}
}
```

produces this output:
```
Real root file system
---------------------
//.        - isDir:1 isDot:1 isFile:0 mtime:1444036970
//..       - isDir:1 isDot:1 isFile:0 mtime:1444036970


Real nested file system
-----------------------
/tmp/.     - isDir:1 isDot:1 isFile:0 mtime:1444054141
/tmp/..    - isDir:1 isDot:1 isFile:0 mtime:1444036970


VFS root file system
--------------------
vfs://root/. - isDir:1 isDot:1 isFile:0 mtime:1444054574
vfs://root/.. - isDir:0 isDot:1 isFile:0 mtime:?
ERROR:  No such file or directory: vfs://root/..


VFS nested file system
----------------------
vfs://root/tmp/. - isDir:1 isDot:1 isFile:0 mtime:1444054574
vfs://root/tmp/.. - isDir:1 isDot:1 isFile:0 mtime:1444054574
```

This is with vfsStream 1.5.0 on PHP 5.5.9 on Ubuntu 14.04.

**Comment by mikey179** [comment]
Just released 1.6.0. Happy migration! :smiley:

**Comment by acoulton** [comment]
Brilliant, thanks very much :)
